File: Ecosystem/Predator.m
classdef Predator < Creature
    %Creature which hunts other creatures
    properties
       
    end
    methods
        function obj = Predator(location)
            obj.location = location;
            obj.regen = 0.05;
            obj.damage = 0.1;
            obj.type = "predator";
            obj.enemy_type = "herbivore";
            obj.food_type = "carcass";
            obj.stress_thresh = 0.85;
            obj.aggressive_placid = .80;
        end
        
    end
end

File: Ecosystem_Python/predator.py
from creature import Creature


class Predator(Creature):
    """Creature which hunts other creatures"""
    
    def __init__(self, location):
        super().__init__(location)
        self.type = "predator"
        self.enemy_type = "herbivore"
        self.food_type = "carcass"
        self.regen = 0.05
        self.damage = 0.1
        self.stress_thresh = 0.85
        self.aggressive_placid = 0.80